File: CrackingTheCodingInterviews/Arrays/OnePointOne.java
package CrackingTheCodingInterviews.Arrays;

import java.util.zip.Checksum;

class OnePointOne {


    // string has all unique characters 

    private static boolean checkString(String target) {

        char check[] = new char[26];

        for (int i=0; i<target.length(); i++) {
            check[target.charAt(i)-'a']++;
            if (check[target.charAt(i)-'a'] > 1) {
                return false;
            }

        }

        return true;
    }


    public static void main(String ar[]) {
        System.out.println(checkString("aa"));
        System.out.println(checkString("abc"));
    }
}

File: CrackingTheCodingInterviews/Arrays/OnePointOne.py
# check if string has duplicate
class OnePointOne:
    def __init__(self) -> None:
        pass

    def check(self, target):
        target = sorted(target)        
        prev = None
        for char in target:
            if prev:
                if prev == char:
                    return True
            prev = char

        return False

one = OnePointOne()
print(one.check("abc"))
print(one.check("aabc"))
print(one.check("abcaaa"))